ml_tools/writers.py

```diff
@@ -156,6 +169,10 @@ def write_figures(self, step: int, figures: Mapping[str, plt.Figure]):
         for w in self._writers:
             w.write_figures(step, figures)
 
+    def write_data(self, data: Mapping[str, Union[Array, Dict]], step: int = None):
```

Comment:
For consistency with the rest of the project, can we use `Mapping` instead of `Dict`?
```suggestion
    def write_data(self, data: Mapping[str, Union[Array, Mapping]], step: int = None):
```